### PR TITLE
Get rid of `verify-no-string-interpolation-leftovers` script in favor of a (hacked) ArchUnit test

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -13,8 +13,8 @@ plugins {
   alias(libs.plugins.versionsFilter)
 }
 
-fun getFlagsForAddOpens(vararg packages: String, module: String): List<String> {
-  return packages.toList().map { "--add-opens=$module/$it=ALL-UNNAMED" }
```
Comment (PawelLipski) [inline build.gradle.kts:17]: `opens` is a superset of `exports`, wasn't needed here (https://stackoverflow.com/a/44059335)

File: src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java [hunk 2]
```diff
@@ -184,4 +186,39 @@ public boolean test(JavaMethodCall javaMethodCall) {
...
+    Method getConstantPool = Class.class.getDeclaredMethod("getConstantPool");
+    getConstantPool.setAccessible(true);
+
+    noClasses()
+        .should(new ArchCondition<>("contain unprocessed string interpolations") {
+          @SneakyThrows
+          @Override
+          public void check(JavaClass clazz, ConditionEvents events) {
+            val constantPool = getConstantPool.invoke(clazz.reflect());
+            int constantPoolSize = (int) constantPool.getClass().getDeclaredMethod("getSize").invoke(constantPool);
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java:202]: Haxxx 

image

File: .circleci/config.yml [hunk 3]
```diff
@@ -84,9 +84,6 @@ jobs:
           # Given the RAM limits on CI (8GB), max-workers=7 cause OOMs.
           # Having 5 or 6 workers doesn't seem to decrease compilation time significantly, as compared to just 4.
           command: ./gradlew --max-workers=4 compileJava compileKotlin
-      - run:
-          name: Run post-compile checks
```
Comment (mkondratek) [inline .circleci/config.yml:88]: is there any significant difference on the pipeline duration?

File: docker/Dockerfile [hunk 4]
```diff
@@ -34,13 +34,12 @@ RUN apt-get install --no-install-recommends -y imagemagick zip
...
 # Secondary packages needed in just one (or few) steps of the pipeline:
 # (package       => needed for command(s))
-# binutils       => strings
 # jq             => jq
 # shellcheck     => shellcheck
 # sudo           => sudo
 # unzip          => zipinfo
 # xxd            => xxd
-RUN apt-get install --no-install-recommends -y binutils jq shellcheck sudo unzip xxd
+RUN apt-get install --no-install-recommends -y jq shellcheck sudo unzip xxd
```
Comment (mkondratek) [inline docker/Dockerfile:42]: _[Jedna cholerna bestia mniej!](https://soundofgothic.pl/record/2/SVM_5_MonsterKilled)_

**Review by mkondratek — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (PawelLipski) [reply .circleci/config.yml:88]: Nopes, it's just a matter of getting stuff more unified

**Pull request merged**